test/test_dataset.py

```diff
@@ -76,50 +94,162 @@ def tearDown(self):
                 except CartoException:
                     warnings.warn('Error deleting tables')
 
+    def test_dataset_constructor_validation(self):
```

Comment:
I'd divide this one in five tests:

- test_dataset_with_table_name_and_query_validation
- test_dataset_with_table_name_and_dataframe_validation
- etc.